File: lua/robot-name/robot-name.lua
local Robot = {}

function Robot:new()
    local r = {}
    setmetatable(r, self)
    self.__index = self
    r:reset()
    return r
end

function Robot:reset()
    local A = 65
    local Z = 90
    self.name = utf8.char(math.random(A, Z)) .. utf8.char(math.random(A, Z)) ..
            math.random(0, 9) .. math.random(0, 9) .. math.random(0, 9)
    return self
end

return Robot

File: nim/robot-name/robot_name.nim
import random

randomize()

type Robot* = object
  name*: string

proc generateName(): string =
  for i in 0..1: result &= sample({'A'..'Z'})
  for i in 0..2: result &= $sample({1..9})

proc makeRobot*(): Robot = Robot(name: generateName())

proc reset*(r: var Robot) = r.name = generateName()
